src/main/java/com/google/devtools/build/lib/remote/http/IdleTimeoutHandler.java

```diff
@@ -29,20 +32,59 @@
   private final TimeoutException timeoutException;
   private boolean closed;
 
+  public final static class WriteTimeoutReaderIdleException extends IOException {
+    public static final WriteTimeoutReaderIdleException INSTANCE = new WriteTimeoutReaderIdleException();
+    private WriteTimeoutReaderIdleException() {}
+  }
+
+  public final static class WriteTimeoutWriterIdleException extends IOException {
+    public static final WriteTimeoutWriterIdleException INSTANCE = new WriteTimeoutWriterIdleException();
+    private WriteTimeoutWriterIdleException() {}
+  }
+
+  public final static class ReadTimeoutReaderIdleException extends IOException {
+    public static final ReadTimeoutReaderIdleException INSTANCE = new ReadTimeoutReaderIdleException();
+    private ReadTimeoutReaderIdleException() {}
+  }
+
+  public final static class ReadTimeoutWriterIdleException extends IOException {
+    public static final ReadTimeoutWriterIdleException INSTANCE = new ReadTimeoutWriterIdleException();
+    private ReadTimeoutWriterIdleException() {}
+  }
+
   @SuppressWarnings("GoodTime-ApiWithNumericTimeUnit")
   public IdleTimeoutHandler(long timeoutSeconds, TimeoutException timeoutException) {
-    super(/* readerIdleTime= */ 0, /* writerIdleTime= */ 0, timeoutSeconds, TimeUnit.SECONDS);
+    super(timeoutSeconds, timeoutSeconds, timeoutSeconds, TimeUnit.SECONDS);
```

Comment:
what's going to change if passing `timeoutSeconds` to `readerIdleTime` and `writerIdleTime`?

Reply:
It makes the corresponding events get fired when the timeout kicks in.